drivers/wifi/esp/esp.c

```diff
@@ -228,20 +228,27 @@ MODEM_CMD_DEFINE(on_cmd_cipsta)
 
 static void esp_ip_addr_work(struct k_work *work)
 {
-	struct esp_data *data = CONTAINER_OF(work, struct esp_data,
-					     ip_addr_work);
+	struct esp_data *dev = CONTAINER_OF(work, struct esp_data,
+					    ip_addr_work);
+	int ret;
+
 	struct modem_cmd cmds[] = {
 		MODEM_CMD("+"_CIPSTA":", on_cmd_cipsta, 2U, ":"),
 	};
 
-	modem_cmd_send(&data->mctx.iface, &data->mctx.cmd_handler,
-			cmds, ARRAY_SIZE(cmds), "AT+"_CIPSTA"?",
-			&data->sem_response, ESP_CMD_TIMEOUT);
+	ret = modem_cmd_send(&dev->mctx.iface, &dev->mctx.cmd_handler,
+			     cmds, ARRAY_SIZE(cmds), "AT+"_CIPSTA"?",
+			     &dev->sem_response, ESP_CMD_TIMEOUT);
+	if (ret < 0) {
+		k_delayed_work_submit_to_queue(&dev->workq, &dev->ip_addr_work,
```

Comment:
Would be nice to have some `LOG_WRN()` or `LOG_ERR()` here.

Reply:
Main reason to have this log is to find out how frequently does AT+CIPSTA? fail. Then we can better tweek or export 5s timeout as Kconfig, as @jukkar proposed.